.github/workflows/test.yaml

```diff
@@ -286,7 +296,8 @@ jobs:
           pip install `echo packages/solara-enterprise/dist/*.whl`[ssg,auth]
           pip install "jupyterlab<4"  "pydantic<2" "playwright==1.41.2" "ipywidgets~=${{ matrix.ipywidgets }}"
           pip freeze --exclude solara --exclude solara-ui --exclude solara-server --exclude pytest-ipywidgets --exclude solara-enterprise > ${{ env.LOCK_FILE_LOCATION }}
-          git diff --quiet || echo "HAS_DIFF=true" >> "$GITHUB_OUTPUT"
+          { git diff --exit-code; echo $? > diff_exit_code.txt; } | tee diff-integration-os${{ matrix.os }}-python${{ matrix.python-version }}-ipywidgets${{ matrix.ipywidgets_major }}.txt;
+          exit_code=$( tail -n 1 diff_exit_code.txt ) | [ exit_code -eq 0 ] || echo "HAS_DIFF=true" >> "$GITHUB_OUTPUT"
```

Comment:
Whoops, forgot some changes. Anyway, the final version is something like:

```bash
{ git diff --exit-code; echo $? > diff_exit_code.txt } | tee diff.txt;
exit_code=$( tail -n 1 diff_exit_code.txt )
[ "$exit_code" -eq 0 ] || echo "HAS_DIFF=true" >> "$GITHUB_OUTPUT"
```

As I understand it, the complexity is because we need to access both the exit code and the the content of `git diff`. The content should go into a file and get printed out, which we handle by using `tee`. However, `tee` never fails, and therefore masks the exit code of `git diff` in the pipe, so we need to write and read that separately.

Reply:
Another way to do this would be to check whether `diff.txt` is empty, but that didn't look like it would turn out simpler (at least to me)

Reply:
I implemented the second method. One thing to note is that if there is a diff, but writing it to a file fails, this will behave as if there was no diff, which could be unexpected.

Reply:
yeah, i still need to ask an llm what [-s file] was, but this seems easier to understand a month from now!